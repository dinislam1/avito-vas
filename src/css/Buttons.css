.button__export button {
  margin: 0 6px !important;
  font-size: 13px !important;
}
