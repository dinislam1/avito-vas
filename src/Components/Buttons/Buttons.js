import React from "react";
import "css/Buttons.css";

export function Buttons(state) {
  let { bgColor, fontSize, fontFamily, text, pastedURL, link } =
    state.init || {};

  const handleSaveImg = () => {
    let canvas = document.getElementById("canvas");
    const dataURL = canvas.toDataURL({
      width: canvas.width,
      height: canvas.height,
      left: 0,
      top: 0,
      format: "png",
    });
    const link = document.createElement("a");
    link.download = "image.png";
    link.href = dataURL;
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
  };

  const handleCopyHTML = () => {
    let canvas = document.getElementById("canvas");
    let dataURL = canvas.toDataURL("image/png");
    navigator.clipboard.writeText(
      `<a href=${link}><img src="${dataURL}"/></a>`
    );
  };

  const handleCopyJSON = () => {
    let json = {};
    json.backgroundColor = bgColor?.hex;
    json.fontSize = fontSize;
    json.fontFamily = fontFamily;
    json.text = text;
    json.imageUrl = pastedURL;
    json.link = link;
    let jsonString = JSON.stringify(json);
    navigator.clipboard.writeText(jsonString);
  };

  return (
    <div className="button__export d-flex my-2">
      <button
        type="button"
        className="btn btn-success btn-save"
        onClick={handleSaveImg}
      >
        Сохранить как png
      </button>
      <button
        type="button"
        className="btn btn-success btn-copy"
        onClick={handleCopyHTML}
      >
        Скопировать тег
      </button>
      <button
        type="button"
        className="btn btn-success btn-json"
        onClick={handleCopyJSON}
      >
        Скопировать JSON
      </button>
    </div>
  );
}
