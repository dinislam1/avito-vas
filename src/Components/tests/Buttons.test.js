import React from "react";
import { shallow, mount, render } from "./enzyme";
import { Buttons } from "Components/Buttons/Buttons";

import renderer from "react-test-renderer";

const setUp = (props) => shallow(<Buttons {...props} />);

describe("Buttons Test Suite", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should contain .button__export wrapper", () => {
    const wrapper = component.find(".button__export");
    expect(wrapper.length).toBe(1);
  });

  it("should contain export buttons", () => {
    const wrapper = component.find("button");
    expect(wrapper.length).toBe(3);
  });

  it("should save image", () => {
    navigator.clipboard = { writeText: jest.fn() };
    const btn = component.find(".btn-json");
    btn.simulate("click");
    const tree = renderer.create(<Buttons />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// describe("Buttons component", () => {
//   it("should render Buttons component", () => {
//     const tree = renderer.create(<Buttons />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
