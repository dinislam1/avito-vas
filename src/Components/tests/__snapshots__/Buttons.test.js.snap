// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Test Suite should save image 1`] = `
<div
  className="button__export d-flex my-2"
>
  <button
    className="btn btn-success btn-save"
    onClick={[Function]}
    type="button"
  >
    Сохранить как png
  </button>
  <button
    className="btn btn-success btn-copy"
    onClick={[Function]}
    type="button"
  >
    Скопировать тег
  </button>
  <button
    className="btn btn-success btn-json"
    onClick={[Function]}
    type="button"
  >
    Скопировать JSON
  </button>
</div>
`;
